src/components/Controls.vue

```diff
@@ -34,36 +34,48 @@
 </template>
 
 <script setup>
-	import { ref } from 'vue'
-	
-	const currentYear = new Date().getFullYear()
-	const selectedYear = ref(currentYear)
+	import { ref, watch } from 'vue'
+
+	const emit = defineEmit(["changeUsername", "changeYear"])
+
 	const props = defineProps({
-		changeUsername: Function,
-		changeYear: Function,
 		lowestYear: {
 			type: Number,
 			default: 2011
 		}
 	})
-
+	
+	const currentYear = new Date().getFullYear()
 	const urlParams = new URLSearchParams(window.location.search)
+
+	const selectedYear = ref(currentYear)
 	const username = ref(urlParams.get("u"))
 
-	if (username.value != null) {
-		props.changeUsername(username.value)
+	if (username != null) {
+		emit("changeYear", username)
 	}
 
 	const range = (start, end) => {
 		if (selectedYear.value < end) {
-			props.changeYear(currentYear)
+			emit("changeYear", currentYear)
```

Comment:
`selectedYear` is now being used as the source of truth on which year is selected (previously just used to set start value, then never really exported again). I think we should set `selectedYear = currentYear `, then emit `selectedYear` here.

Reply:
ok cool